src/main/java/cz/cvut/kbss/jsonld/deserialization/expanded/ExpandedJsonLdDeserializer.java

```diff
@@ -29,14 +29,25 @@
 import jakarta.json.JsonArray;
 import jakarta.json.JsonObject;
 import jakarta.json.JsonValue;
+import java.util.HashMap;
+import java.util.Map;
 
 public class ExpandedJsonLdDeserializer extends JsonLdDeserializer {
 
+	private final Map<String, Object> knownInstances;
+	final PendingReferenceRegistry referenceRegistry;
+
     public ExpandedJsonLdDeserializer() {
+		deserializers.configure(configuration());
+		knownInstances = new HashMap<>();
+		referenceRegistry = new PendingReferenceRegistry();
```

Comment:
I think these need to be reset before/after each deserialization call. The `ExpandedJsonLdDeserializer` is not a one-of instance that is thrown away after each deserialization and this could transfer state between calls. If my understanding is correct, they could be reset:
A) After deserialization if `disableUnresolvedReferencesCheck` is **not** configured
B) In `cleanup` if `disableUnresolvedReferencesCheck` is set